src/v2.ts

```diff
@@ -0,0 +1,141 @@
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import {
+  CloudFunction,
+  CloudEvent,
+} from 'firebase-functions/v2';
+
+/** A function that can be called with test data and optional override values for {@link CloudEvent}
+ * It will subsequently invoke the cloud function it wraps with the provided {@link CloudEvent}
+ */
+export type WrappedFunction = (
+  cloudEvent?: CloudEvent
+) => any | Promise<any>;
+
+/** A scheduled function that can be called with a {@link CloudEvent}.
+ * It will subsequently invoke the cloud function it wraps with a generated {@link CloudEvent}.
+ */
+export type WrappedScheduledFunction = (
+  cloudEvent?: CloudEvent
+) => any | Promise<any>;
+
+/** Takes a v2 cloud function to be tested, and returns a {@link WrappedFunction}
+ * which can be called in test code.
+ */
+export function wrap<T>(
+  cloudFunction: CloudFunction<T>
+): WrappedScheduledFunction | WrappedFunction {
+
+  // TODO(tystark) Verify this is the appopriate handle for ScheduledFunctions
+  if (
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels &&
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels['deployment-scheduled'] === 'true') {
+    return (cloudEvent: CloudEvent): WrappedScheduledFunction => cloudFunction.run(cloudEvent);
+  }
+
+  // TODO(tystark) verify
+  if (
+    // @ts-ignore
+    !!(cloudFunction?.__trigger?.httpsTrigger) &&
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels &&
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels['deployment-callable'] !== 'true'
+  ) {
+    throw new Error(
+      'Wrap function is only available for `onCall` HTTP functions, not `onRequest`.'
+    );
+  }
+
+  if (!cloudFunction.run) {
+    throw new Error(
+      'This library can only be used with functions written with firebase-functions v1.0.0 and above'
+    );
+  }
+
+  // TODO(tystark) verify
+  if (!cloudFunction.__endpoint) {
+    throw new Error(
+      'This function can only wrap V2 CloudFunctions.'
+    );
+  }
+
+  let wrapped: WrappedFunction = (cloudEvent: CloudEvent) => {
+    _checkCloudEventValidity(
+      [
+        'specversion',
+        'id',
+        'source',
+        'subject',
+        'type',
+        'time',
+        'data',
+        'params'],
+      cloudEvent);
+
+    return cloudFunction.run(cloudEvent);
+  };
+
+  return wrapped;
+}
+
+/** @internal */
+
+/** @return CloudEvent populated with default values */
+export const _getDefaultCloudEvent = (): CloudEvent => (
```

Comment:
You're probably going to need a default cloud event per service provider (to mock data) and it's going to need to take the function's `__endpoint` into account to fill in details (e.g. source, subject, and type)

Reply:
Ack, I update that part shortly